trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1302,10 +1302,19 @@ where
         let response_clone = response.clone();
 
         tokio::spawn(async move {
-            let mut conn = rx.await.unwrap();
+            let mut conn = match rx.await {
+                Ok(conn) => conn,
+                Err(err) => {
+                    error!(error = %err, "Error receiving UTP stream from UtpListener");
+                    return;
+                }
+            };
             // Handle STATE packet for SYN
             let mut buf = [0; BUF_SIZE];
-            conn.recv(&mut buf).await.unwrap();
+            if let Err(err) = conn.recv(&mut buf).await {
+                error!(error = %err, "Error receiving UTP stream from UtpListener");
```

Comment:
It's almost always helpful to have the string be unique. To make this different from 1308:
```suggestion
                error!(error = %err, "Error receiving STATE packet from UtpListener");
```